karabo_data/reader.py

```diff
@@ -653,21 +653,44 @@ def select(self, seln_or_source_glob, key_glob='*'):
 
         There are three possible ways to select data:
 
-        1. With glob patterns (* is a wildcard) for source and key::
+        1. With two glob patterns (see below) for source and key names::
 
+            # Select data in the image group for any detector sources
             sel = run.select('*/DET/*, 'image.*')
 
         2. With a list of (source, key) glob patterns::
 
+            # Select image.data and image.mask for any detector sources
             sel = run.select([('*/DET/*, 'image.data'), ('*/DET/*, 'image.mask')])
 
+           Data is included if it matches any of the pattern pairs.
+
         3. With a dict of source names mapped to sets of key names
            (or empty sets to get all keys)::
 
+            # Select image.data from one detector source, and all data from one XGM
             sel = run.select({'SPB_DET_AGIPD1M-1/DET/ALLCH:xtdf': {'image.data'},
                               'SA1_XTD2_XGM/XGM/DOOCS': set()})
 
+           Unlike the others, this option *doesn't* allow glob patterns.
+           It's a more advanced option for code that knows precisely what
```

Comment:
What does "more advanced" means here? is it faster to use it that way?
Also the `ALLCH:xtdf` in the example confuses me a bit... is that a thing?

Reply:
It's more advanced in the sense that it's precise - you'll never unexpectedly get sources that you didn't mean to select - but less convenient, because you have to spell out the source names you want in full.

I think the ALLCH bit must have been copied across from a karabo-bridge example at some point - it's not a thing in the files, as far as I know. I'll change that.

Reply:
(This option is likely to be marginally faster, because the other options are expanded into this structure, but the difference should be small enough not to matter)